### DISCUSSION TITLE
Immutable objects

### DISCUSSION THREAD

**Discussion by TheManFran** [ngrx/platform#4242] [Q&A]
Hi

Has something changed in Ngrx 17 around immutability. Code that worked before I upgraded worked, but now I get errors that I can't modify an object which is read only. The object that I can't modify is the one I send in an action, and store in my state. Almost like in the past the reference to the original object was broken, and now it's not.

**Comment by timdeschryver** [comment]
This isn't something that is changed in v17, it should still work as in v16.
@TheManFran can you provide a repository with a reproduction?

**Reply by timdeschryver** [reply]
This could be related to #4237
